janet/src/main/java/io/techery/janet/ActionServiceWrapper.java

```diff
@@ -45,22 +46,29 @@ public ActionServiceWrapper(ActionService actionService) {
      * Called from service callback before changing action status to {@linkplain ActionState.Status#SUCCESS SUCCESS}
      *
      * @param holder action holder for intercepting
+     * @return if {@code true} action is finished without any statuses and won't be sent to callback with status {@linkplain ActionState.Status#SUCCESS SUCCESS}
      */
-    protected abstract <A> void onInterceptSuccess(ActionHolder<A> holder);
+    protected abstract <A> boolean onInterceptSuccess(ActionHolder<A> holder);
 
     /**
      * Called from service callback before changing action status to {@linkplain ActionState.Status#FAIL FAIL}
      *
      * @param holder action holder for intercepting
+     * @return if {@code true} action will be sent again. Should be careful with it because there is possibility to create an infinite loop of action sending
      */
-    protected abstract <A> void onInterceptFail(ActionHolder<A> holder, JanetException e);
+    protected abstract <A> boolean onInterceptFail(ActionHolder<A> holder, JanetException e);
 
     /**
      * {@inheritDoc}
      */
     @Override protected <A> void sendInternal(ActionHolder<A> holder) throws JanetException {
         if (onInterceptSend(holder)) callback.onSuccess(holder);
-        else actionService.sendInternal(holder);
+        else try {
+            actionService.sendInternal(holder);
+        } catch (JanetException e) {
+            this.callback.onFail(holder, e);
```

Comment:
yes we really need it because `sendInternal` can throw excaption and we have to handle it. 
it's because of we override method `sendInternal` instead of `send` where is throw catching and callback calling.